ochk/resource_virtual_machine.go

```diff
@@ -0,0 +1,314 @@
+package ochk
+
+import (
+	"context"
+	"fmt"
+	"github.com/chmurakrajowa/terraform-provider-ochk/ochk/sdk"
+	"github.com/chmurakrajowa/terraform-provider-ochk/ochk/sdk/gen/models"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	VirtualMachineRetryTimeout = 20 * time.Minute
+)
+
+func resourceVirtualMachine() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVirtualMachineCreate,
+		ReadContext:   resourceVirtualMachineRead,
+		UpdateContext: resourceVirtualMachineUpdate,
+		DeleteContext: resourceVirtualMachineDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(VirtualMachineRetryTimeout),
+			Update: schema.DefaultTimeout(VirtualMachineRetryTimeout),
+			Delete: schema.DefaultTimeout(VirtualMachineRetryTimeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"deployment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"initial_password": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
co to robi? Może przydałby się jakiś komentarz dla potomnych, bo wygląda trochę magicznie ;)

Reply:
done